spec/requests/api/v1/moves_controller_filter_spec.rb

```diff
@@ -27,6 +27,30 @@
           expect(response_json['data'].map { |move| move['id'] }.sort).to eq(location.moves_from.pluck(:id).sort)
         end
       end
+
+      describe 'by created_at' do
+        let(:first_date) { Date.new(2019, 12, 25) }
+        let(:last_date) { Date.new(2019, 12, 27) }
+        let(:middle_date) { Date.new(2019, 12, 26) }
+
+        let(:target_move) { Move.find_by(created_at: first_date) }
+        let(:target_move2) { Move.find_by(created_at:  middle_date) }
+        let(:target_move3) { Move.find_by(created_at:  last_date) }
+        let(:filter_params) { { filter: { created_at_from: first_date.to_s, created_at_to: last_date.to_s } } }
+
+        before do
+          create(:move, created_at: first_date)
+          create(:move, created_at: middle_date)
+          create(:move, created_at: last_date)
+
+          get '/api/v1/moves', headers: headers, params: filter_params
+        end
+
+        it 'returns the right amount of moves' do
+          expect(response_json['data'].map { |x| x['attributes']['created_at'] }).
+            to match_array [first_date, middle_date, last_date].map(&:to_datetime).map(&:iso8601)
+        end
```

Comment:
that's very dense,  can you think something less "smart" :) 
it would make more readable

Reply:
less 'smart'? I'm asserting that we get back the data we expect, not sure how to improve it?